### PR TITLE
chore(external docs): Fill in missing parts of the documentation at the kubernetes_logs source

### LINKED ISSUE #4587: Add documentation for the kubernetes_logs source on using custom selectors
A follow up from https://github.com/timberio/vector/pull/4182#issuecomment-709482798.

### LINKED ISSUE #4594: Finish Kubernetes reference docs
See https://github.com/timberio/vector/pull/4593

### PR DESCRIPTION
**Pull request by MOZGIII**
This PR fills in some details at the `kubernetes_logs` source docs.

@binarylogic, could you do a pass on making the values fit the schema? I'm having a hard time figuring out the error cause, as errors are in pretty much all of the components.

Closes #4594.
Closes #4587.

### CODE REVIEW

**Comment by MOZGIII** [comment]
Converting back to draft, will add new params from #4586 and open for reviews again.

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 1]
```diff
@@ -126,13 +137,167 @@ components: sources: kubernetes_logs: {
...
+					"pod_labels": {
+						"addonmanager.kubernetes.io/mode": "Reconcile"
+						"gcp-auth-skip-secret":            "true"
+						"integration-test":                "storage-provisioner"
+					}
+					"pod_name":      "storage-provisioner"
+					"pod_namespace": "kube-system"
+					"pod_node_name": "minikube"
+					"pod_uid":       "93bde4d0-9731-4785-a80e-cd27ba8ad7c2"
+				}
```
Comment (binarylogic) [inline docs/reference/components/sources/kubernetes_logs.cue:292]: Should these be nested or should they be top-level keys? Ex: `"kuberenetes.container_image"`?

Comment (MOZGIII) [reply docs/reference/components/sources/kubernetes_logs.cue:292]: Right, this was just copied from the `stdout` sink output, I'll correct this. Good catch!

**Comment by jamtur01** [comment]
@MOZGIII Looks like there's still some Cue failures here?

**Pull request merged**